h2o-k8s/src/main/java/water/k8s/KubernetesEmbeddedConfigProvider.java

```diff
@@ -0,0 +1,94 @@
+package water.k8s;
+
+import water.init.AbstractEmbeddedH2OConfig;
+import water.init.EmbeddedConfigProvider;
+import water.k8s.lookup.KubernetesDnsLookup;
+import water.k8s.lookup.KubernetesLookup;
+import water.k8s.lookup.LookupConstraintsBuilder;
+import water.util.Log;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * A configuration provider for H2O running in Kubernetes cluster. It is able to detected it's ran in K8S
+ * environment, otherwise remains inactive.
+ * <p>
+ * Uses potentially multiple strategies to discover H2O Pods on a Kubernetes cluster.
+ */
+public class KubernetesEmbeddedConfigProvider implements EmbeddedConfigProvider {
+
+    private static final String K8S_NODE_LOOKUP_TIMEOUT_KEY = "H2O_NODE_LOOKUP_TIMEOUT";
+    private static final String K8S_DESIRED_CLUSTER_SIZE_KEY = "H2O_NODE_EXPECTED_COUNT";
+
+    private boolean runningOnKubernetes = false;
+    private KubernetesEmbeddedConfig kubernetesEmbeddedConfig;
+
+    /**
+     * @return A Set
+     */
+    private static final Optional<Set<String>> resolveNodeIPs() {
+        final LookupConstraintsBuilder lookupConstraintsBuilder = new LookupConstraintsBuilder();
+
+        try {
+            final Integer timeoutSeconds = Integer.parseInt(System.getenv(K8S_NODE_LOOKUP_TIMEOUT_KEY));
+            lookupConstraintsBuilder.withTimeoutSeconds(timeoutSeconds);
+        } catch (NumberFormatException e) {
+            Log.info(String.format("'%s' environment variable not set.", K8S_NODE_LOOKUP_TIMEOUT_KEY));
+        }
+
+        try {
+            final Integer desiredClusterSize = Integer.parseInt(System.getenv(K8S_DESIRED_CLUSTER_SIZE_KEY));
+            lookupConstraintsBuilder.withDesiredClusterSize(desiredClusterSize);
+        } catch (NumberFormatException e) {
+            Log.info(String.format("'%s' environment variable not set.", K8S_DESIRED_CLUSTER_SIZE_KEY));
```

Comment:
we could mention that we are falling back to some default value

Reply:
Here, we are only saying environment variable is not set. The actual lookup constraint logic says exactly that. It will tell you if it falls back to a default timeout.